src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpectatorView.cs

```diff
@@ -206,5 +238,70 @@ private bool TryCreateRecordingService(out IRecordingService recordingService)
+            {
+                UnityEngine.Debug.Log($"SpatialLocalizationInitializationSettings: {message}");
+            }
+        }
+
+        private async void OnParticipantConnected(SpatialCoordinateSystemParticipant participant)
+        {
+            DebugLog($"Waiting for the set of supported localizers from connected participant {participant.SocketEndpoint.Address}");
+
+            // When a remote participant connects, get the set of ISpatialLocalizers that peer
+            // supports. This is asynchronous, as it comes across the network.
+            ISet<Guid> peerSupportedLocalizers = await participant.GetPeerSupportedLocalizersAsync();
+
+            // If there are any supported localizers, find the first configured localizer in the
+            // list that supports that type. If and when one is found, use it to perform localization.
+            if (peerSupportedLocalizers != null)
+            {
+                DebugLog($"Received a set of {peerSupportedLocalizers.Count} supported localizers");
+                
+                if (SpatialLocalizationInitializationSettings.IsInitialized &&
+                    TryRunLocalization(SpatialLocalizationInitializationSettings.Instance.PrioritizedInitializers, peerSupportedLocalizers, participant))
+                {
+                    // Succeeded at using a custom localizer specified by the app.
+                    return;
+                }
+
+                if (TryRunLocalization(defaultSpatialLocalizationInitializers, peerSupportedLocalizers, participant))
+                {
+                    // Succeeded at using one of the default localizers from the prefab.
+                    return;
+                }
+
+                if (defaultSpatialLocalizationInitializers != null)
+
+                DebugLog($"None of the configured LocalizationInitializers were supported by the connected participant, localization will not be started");
+            }
+            else
+            {
+                DebugLog($"No supported localizers were received from the participant, localization will not be started");
+            }
+        }
+
+        private bool TryRunLocalization(IList<SpatialLocalizationInitializer> initializers, ISet<Guid> supportedLocalizers, SpatialCoordinateSystemParticipant participant)
+        {
+            if (initializers == null)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < initializers.Count; i++)
+            {
+                if (supportedLocalizers.Contains(initializers[i].PeerSpatialLocalizerId))
+                {
+                    DebugLog($"Localization initializer {initializers[i].GetType().Name} supported localization with ID {initializers[i].PeerSpatialLocalizerId}, starting localization");
+                    initializers[i].RunLocalization(participant);
+                    return true;
+                }
+            }
+
+            return false;

```

Comment:
>  return false; [](start = 10, length = 15)

Consider printing a warning saying: "We have x registered, but none is supported for {id}"

Reply:
The equivalent of that log message happens in OnParticipantConnected if neither the user-customized nor built-in defaults worked.

---
In reply to: [298335757](https://github.com/microsoft/MixedReality-SpectatorView/pull/48#discussion_r298335757) [](ancestors = 298335757)